plugins/inputs/statsd/statsd.go

```diff
@@ -307,6 +312,9 @@ func (s *Statsd) Gather(acc telegraf.Accumulator) error {
 		fields := make(map[string]interface{})
 		for field, set := range m.fields {
 			fields[field] = int64(len(set))
+			if s.FloatSets {
+				fields[field] = float64(len(set))
+			}
```

Comment:
I think this is cleaner and easier to read
```suggestion
			if s.FloatSets {
				fields[field] = float64(len(set))
			} else {
				fields[field] = int64(len(set))
			}
```